ValheimPlus/GameClasses/InventoryGUI.cs

```diff
@@ -104,4 +108,77 @@ public static void Postfix(ref InventoryGui __instance)
 			}
 		}
 	}
+
+	[HarmonyPatch(typeof(InventoryGui), nameof(InventoryGui.RepairOneItem))]
+	public static class InventoryGui_RepairOneItem_Transpiler
+	{
+		private static MethodInfo method_EffectList_Create = AccessTools.Method(typeof(EffectList), nameof(EffectList.Create));
+		private static MethodInfo method_CreateNoop = AccessTools.Method(typeof(InventoryGui_RepairOneItem_Transpiler), nameof(InventoryGui_RepairOneItem_Transpiler.CreateNoop));
+
+		/// <summary>
+		/// Patches out the code that spawns an effect for each item repaired - when we repair multiple items, we only want
+		/// one effect, otherwise it looks and sounds bad. The patch for InventoryGui.UpdateRepair will spawn the effect instead.
+		/// </summary>
+		[HarmonyTranspiler]
+		public static IEnumerable<CodeInstruction> Transpile(IEnumerable<CodeInstruction> instructions)
+		{
+			if (!Configuration.Current.Player.IsEnabled) return instructions;
+
+			List<CodeInstruction> il = instructions.ToList();
+
+			if (Configuration.Current.Player.autoRepair)
+			{
+				// We look for a call to EffectList::Create and replace it with our own noop stub.
+				for (int i = 0; i < il.Count; ++i)
+				{
+					if (il[i].Calls(method_EffectList_Create))
+					{
+						il[i].opcode = OpCodes.Call; // original is callvirt, so we need to tweak it
+						il[i].operand = method_CreateNoop;
+					}
+				}
+			}
+
+			return il.AsEnumerable();
+		}
+
+		private static GameObject[] CreateNoop(Vector3 _0, Quaternion _1, Transform _2, float _3)
+		{
+			return null;
+		}
+	}
+
+	[HarmonyPatch(typeof(InventoryGui), nameof(InventoryGui.UpdateRepair))]
+	public static class InventoryGui_UpdateRepair_Patch
+	{
+		/// <summary>
+		/// When we're in a state where the InventoryGui is open and we have items available to repair,
+		/// and we have an active crafting station, this patch is responsible for repairing all items
+		/// that can be repaired and then spawning one instance of the repair effect if at least one item
+		/// has been repaired.
+		/// </summary>
+		[HarmonyPrefix]
+		public static void Prefix(InventoryGui __instance)
```

Comment:
I think this parameter requires `ref`

Reply:
It would only require ref if `InventoryGui` was a value type. Since `InventoryGui` is a reference type, we can do things to its members without using ref. We would only need a ref if this was, for example, an argument and we wanted to change which object it pointed to, like this:

```cs
[HarmonyPrefix]
public static void Prefix(InventoryGui arg)
{
    arg = null;
}
```

This would do nothing.

```cs
[HarmonyPrefix]
public static void Prefix(ref InventoryGui arg)
{
    arg = null;
}
```

This would change the argument that is passed to the original function to be null.

Reply:
so by not passing `ref` we'll still have the same desired affect when running `__instance.RepairOneItem()`?

Reply:
Yep, in this case, the behaviour would only be different if `InventoryGui` was a value type (in other words, a struct).